provider/lxd/provider.go

```diff
@@ -25,22 +25,48 @@ import (
 	"github.com/juju/juju/provider/lxd/lxdnames"
 )
 
+// ProviderLXDServer provides methods for the Provider and the
+// ProviderCredentials to query.
+//go:generate mockgen -package lxd -destination provider_mock_test.go github.com/juju/juju/provider/lxd ProviderLXDServer
+type ProviderLXDServer interface {
```

Comment:
Initially, I intended to avoid shimming out our implementation of ```Server``` and rely on the mocks of the upstream LXD interfaces, but this is probably going to make tests clearer.

Reply:
Yeah, I think we should own the things that we use, mainly for consistency and to better understand our dependencies.